src/testWithSpringSecurity_5_7/java/org/openrewrite/spring/security5/UpgradeSpringSecurity61Test.java

```diff
@@ -0,0 +1,164 @@
+              import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+              import org.springframework.security.web.SecurityFilterChain;
+
+              @EnableWebSecurity
+              class SecurityConfig extends WebSecurityConfigurerAdapter {
+                  @Override
+                  public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+                      http
+                              .authorizeHttpRequests(requests -> requests
+                                      .requestMatchers(HttpMethod.OPTIONS, "/rest/**").permitAll()
+                                      .anyRequest().permitAll()).csrf(csrf -> csrf.disable());
+                      return http.build();
+                  }
+              }
+              """,
+            // TODO Remove this diagnostic
+            spec -> spec.afterRecipe(cu -> {
+                RecipesThatMadeChanges recipesThatMadeChanges = cu.getMarkers().findFirst(RecipesThatMadeChanges.class).get();
+                Collection<List<Recipe>> recipes = recipesThatMadeChanges.getRecipes();
+                for (List<Recipe> recipeList : recipes) {
+                    for (Recipe recipe : recipeList) {
+                        System.out.printf(recipe.getName() + "\n");
+                    }
+                    System.out.printf("\n");
+                }
+            })
+          )
+        );
+    }
+
+    @Test
+    void shouldRetainChainOfAntMatchers() {
+        //language=java
+        rewriteRun(
+          spec -> spec.expectedCyclesThatMakeChanges(2),
+          java(
+            """
+              import org.springframework.http.HttpMethod;
+              import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+              import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+              import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+              import org.springframework.security.web.SecurityFilterChain;
+
+              @EnableWebSecurity
+              class SecurityConfig extends WebSecurityConfigurerAdapter {
+                  @Override
+                  public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+                      http
+                          .authorizeRequests()
+                          .antMatchers(HttpMethod.OPTIONS, "/rest/**").permitAll()
+                          .antMatchers("/openapi").permitAll()
+                          .antMatchers("/openapi.yaml").permitAll()
+                          .antMatchers("/openapi/**").permitAll()
+                          .antMatchers("/rest/hello").permitAll()
+                          .antMatchers("/actuator/**").permitAll()
+                          .anyRequest().authenticated()
+                          .and().oauth2ResourceServer().jwt();
```

Comment:
Similar to above; can we strip out anything not necessary to replicate the issue? Eyeing the `oauth2ResourceServer` here, possibly others.